torchgeo/samplers/single.py

```diff
@@ -142,10 +156,16 @@ def __iter__(self) -> Iterator[BoundingBox]:
             hit = self.hits[idx]
             bounds = BoundingBox(*hit.bounds)
```

Comment:
Only the first hit (file) is chosen. Currently I use this file to extract the footprint. But the sample is read from the merged raster, and the footprint for this one file might not cover the other.

Should be fairly easy to combine the footprints for all hits and send that to `get_random_bounding_box_check_valid_overlap`. 

Let me know if my assumptions are correct, and I will implement it.

Reply:
The footprint for a hit is static. So this could be joined and cropped during init of IntersectionDataset

Reply:
I assume I have understood this correctly, and added this functionality in `_merge_dataset_indices`.